lib/mail_catcher/message.rb

```diff
@@ -0,0 +1,18 @@
+module MailCatcher
+  def send_data(data)
+    Async do
+      endpoint = Async::HTTP::Endpoint.parse("#{MailCatcher.http_url}/messages")
+
+      begin
+        Async::WebSocket::Client.connect(endpoint) do |connection|
+          puts 'Connected...'
+
+          connection.write data
+          connection.flush
+        end
```

Comment:
What's this bit about, is this constructing a websocket client to itself?

Reply:
Yes, I have used [Async Websocket](https://github.com/socketry/async-websocket) gem which needs to create a client to itself to send data. you can see the example [here](https://github.com/socketry/async-websocket/blob/master/examples/rack/client.rb)

Reply:
Ah. I don’t think we should do that. There should be a way to push a message to clients without being a client. And clients probably shouldn’t be able to send each other messages.

Reply:
I think there's no way to push a message without the `connection` object which we get from the connection to WebSocket itself.

Reply:
I've pushed a new version which uses a central bus for broadcasting and ignores client messages as I was describing.